src/Query/Transformer/NamedArrayParameter.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Linio\Component\Database\Query\Transformer;
+
+use Linio\Component\Database\Exception\InvalidQueryException;
+use Linio\Component\Database\Query\Transformer;
+
+class NamedArrayParameter implements Transformer
+{
+    public function execute(string &$query, array &$params = []): void
+    {
+        foreach ($params as $paramKey => $paramValue) {
+            $isNumeric = is_numeric($paramKey);
+            $isArray = is_array($paramValue);
+
+            if ($isNumeric && $isArray) {
+                throw new InvalidQueryException(sprintf(
+                    'Unnamed parameter "%s" can\'t have array value: %s',
```

Comment:
I wouldn't put the value in the exception. This may cause more issues with sensitive information being leaked to logs. Also, we don't really care what the value is, just that it is an array.

Reply:
makes sense. tks.